src/mmg3d/boulep_3d.c

```diff
@@ -514,8 +516,10 @@ int MMG5_boulernm(MMG5_pMesh mesh,MMG5_Hash *hash,MMG5_int start,int ip,MMG5_int
           else if ( ph->a ) {
             while ( ph->nxt && ph->nxt < hash->max ) {
               ph = &hash->item[ph->nxt];
-              if ( ph->a == ia && ph->b == ib )  continue;
```

Comment:
Remark: This erroneous `continue` was leading to add a new edge in the hash table even if this edge was already listed. In consequence, the counted number of  non-manifold edges was sometime greater than the real nomber of nm edges passing through the point, leading to add spurious required vertices.

See the attached example where 2 spurious required vertices are added.

[nm-save-serial.mesh.txt](https://github.com/MmgTools/mmg/files/12106728/nm-save-serial.mesh.txt)